File: pms-backend/src/main/java/com/pms/service/MedicalRecordService.java
package com.pms.service;

import com.pms.dto.MedicalRecordDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface MedicalRecordService {
    MedicalRecordDto createMedicalRecord(MedicalRecordDto dto);
    MedicalRecordDto updateMedicalRecord(Long id, MedicalRecordDto dto);
    MedicalRecordDto getMedicalRecordById(Long id);
    List<MedicalRecordDto> getMedicalRecordsByPatientId(Long patientId);
    List<MedicalRecordDto> getMedicalRecordsByDoctorId(Long doctorId);
    Page<MedicalRecordDto> searchMedicalRecords(Long patientId, Long doctorId, String query, Pageable pageable);
}

File: pms-frontend/src/services/medicalRecordService.js
import api from './api';

export const medicalRecordService = {
  searchMedicalRecords: async (params = {}) => {
    const res = await api.get('/medical-records', { params });
    return res.data;
  },

  getMedicalRecordById: async (id) => {
    const res = await api.get(`/medical-records/${id}`);
    return res.data;
  },

  getMedicalRecordsByPatientId: async (patientId) => {
    const res = await api.get(`/medical-records/patient/${patientId}`);
    return res.data;
  },

  createMedicalRecord: async (data) => {
    const res = await api.post('/medical-records', data);
    return res.data;
  },

  updateMedicalRecord: async (id, data) => {
    const res = await api.put(`/medical-records/${id}`, data);
    return res.data;
  }
};
